packages/xstate-store/src/atom.ts

```diff
@@ -39,6 +40,30 @@ const [
 
 let activeSub: Subscriber | undefined = undefined;
 
+type AsyncAtomState<Data, Error = unknown> =
+  | { status: 'pending' }
+  | { status: 'fulfilled'; data: Data }
+  | { status: 'rejected'; error: Error };
+
+export function createAsyncAtom<T>(
+  getValue: () => Promise<T>,
+  options?: AsyncAtomOptions<AsyncAtomState<T>>
+): ReadonlyAtom<AsyncAtomState<T>> {
+  const atom = createAtom<AsyncAtomState<T>>(() => {
+    getValue()
+      .then((value) => {
+        atom._update(() => ({ status: 'fulfilled', data: value }));
+      })
+      .catch((error) => {
+        atom._update(() => ({ status: 'rejected', error }));
+      });
```

Comment:
nit: I always felt this is more correct - because it "wraps" the most minimal thing with ur catch handler. We only want to catch errors coming from `getValue()` and not the ones coming from the `then`'s callback
```suggestion
      .then((value) => {
        atom._update(() => ({ status: 'fulfilled', data: value }));
      }, (error) => {
        atom._update(() => ({ status: 'rejected', error }));
      })
```